packages/salesforcedx-vscode-lwc/src/commands/lightningLwcPreview.ts

```diff
@@ -144,24 +143,31 @@ export const lwcPreview = async (sourceUri: URI) => {
     return;
   }
 
-  if (!fs.existsSync(resourcePath)) {
-    const message = nls.localize('lightning_lwc_preview_file_nonexist', resourcePath);
-    showError(new Error(message), logName, commandName);
-    return;
-  }
+  try {
+    const stat = await vscode.workspace.fs.stat(sourceUri);
+    if (!stat) {
+      const message = nls.localize('lightning_lwc_preview_file_nonexist', resourcePath);
+      showError(new Error(message), logName, commandName);
+      return;
+    }
 
-  const isSFDX = true; // TODO support non SFDX Projects
-  const isDirectory = fs.lstatSync(resourcePath).isDirectory();
-  const componentName = isDirectory
-    ? componentUtil.moduleFromDirectory(resourcePath, isSFDX)
-    : componentUtil.moduleFromFile(resourcePath, isSFDX);
-  if (!componentName) {
-    const message = nls.localize('lightning_lwc_preview_unsupported', resourcePath);
+    const isSFDX = true; // TODO support non SFDX Projects
+    const isDirectory = stat.type === vscode.FileType.Directory;
+    const componentName = isDirectory
+      ? componentUtil.moduleFromDirectory(resourcePath, isSFDX)
+      : componentUtil.moduleFromFile(resourcePath, isSFDX);
+    if (!componentName) {
+      const message = nls.localize('lightning_lwc_preview_unsupported', resourcePath);
+      showError(new Error(message), logName, commandName);
+      return;
+    }
+
+    await executePreview(startTime, componentName, resourcePath);
+  } catch (error) {
```

Comment:
checking my understanding: if that new `stat` fails, we end up here?

also, you don't have to have the error var if you're not going to actually use it
```suggestion
  } catch {
```